components/gemini/browser/gemini_service.cc

```diff
@@ -115,6 +115,23 @@ namespace {
     return encoded_csrf;
   }
 
+  void CalculateSaleAmount(std::string quantity,
+                           std::string price,
+                           std::string fee,
+                           std::string* total_price) {
+    if (quantity.empty() || price.empty() || fee.empty() || !total_price) {
+      return;
+    }
+
+    double parsed_quantity = std::stod(quantity);
+    double parsed_price = std::stod(price);
+    double parsed_fee = std::stod(fee);
```

Comment:
You might like `base::StringToDouble`, it makes failures a little more obvious because it returns a `bool`.

Reply:
good idea, now using this 👍